R/read_data.R

```diff
@@ -0,0 +1,165 @@
+  mapped_disease <- disease_map[[disease]]
+
+  parameters <- list(
+    data_path,
+    mapped_disease,
+    stringify_date(min_reference_date),
+    stringify_date(max_reference_date),
+    stringify_date(report_date)
+  )
+
+  # We need different queries for the states and the US overall. For US overall
+  # we need to aggregate over all the facilities in all the states. For the
+  # states, we need to aggregate over all the facilities in that one state
+  if (state_abb == "US") {
+    query <- "
+   SELECT
+     sum(value) AS confirm,
+     reference_date,
+     report_date,
+     -- We want to inject the 'US' as our abbrevation here bc data is not agg'd
+     'US' AS state_abb
+    FROM read_parquet(?)
+    WHERE 1=1
+      AND disease = ?
+      AND metric = 'count_ed_visits'
+      AND reference_date >= ?
+      AND reference_date <= ?
+      AND report_date = ?
+    GROUP BY reference_date, report_date
+    ORDER BY reference_date
+   "
+  } else {
+    # We want just one state so aggregate over facilites in that one state only
+    query <- "
+  SELECT
+    sum(value) AS confirm,
+    reference_date,
+    report_date,
+    geo_value AS state_abb
+  FROM read_parquet(?)
+  WHERE 1=1
+    AND disease = ?
+    AND metric = 'count_ed_visits'
+    AND reference_date >= ?
+    AND reference_date <= ?
+    AND report_date = ?
+    AND geo_value = ?
+  GROUP BY geo_value, reference_date, report_date
+  ORDER BY reference_date
+  "
+    # Append `geo_value` to the query
+    parameters <- c(parameters, list(state_abb))
+  }
+
+  # Guard against file does not exist
+  cli::cli_alert("Reading data from {.path {data_path}}")
+  if (!file.exists(data_path)) {
+    cli::cli_abort(
+      "Cannot read data. File {.path {data_path}} doesn't exist",
+      class = "file_not_found"
+    )
+  }
+
+  con <- DBI::dbConnect(duckdb::duckdb())
+  df <- rlang::try_fetch(
+    DBI::dbGetQuery(
+      con,
+      statement = query,
+      params = parameters
+    ),
+    error = function(con) {
+      cli::cli_abort(
+        c(
+          "Error fetching data from {.path {data_path}}",
+          "Using parameters {parameters}",
+          "Original error: {con}"
+        ),
+        class = "wrapped_invalid_query"
+      )
+    }
+  )
+  DBI::dbDisconnect(con)
+
+  # Guard against empty return
+  if (nrow(df) == 0) {
+    cli::cli_abort(
+      c(
+        "No data matching returned from {.path {data_path}}",
+        "Using parameters {parameters}"
+      ),
+      class = "empty_return"
+    )
+  }
+  # Warn for incomplete return
+  if (
+    nrow(df) != as.Date(max_reference_date) - as.Date(min_reference_date) + 1
+  ) {
+    n_rows_expected <- as.Date(max_reference_date) - as.Date(min_reference_date)
+    expected_dates <- format(seq.Date(
+      from = as.Date(min_reference_date),
+      to = as.Date(max_reference_date),
+      by = "day"
+    ), "%Y-%m-%d")
+    missing_dates <- expected_dates[
+      !(expected_dates %in% df[["reference_date"]])
+    ]
```

Comment:
```suggestion
    missing_dates <- setdiff(expected_dates, df[["reference_date"]])
```
Feel free to ignore. This just feels slightly easier.